troubadix/argparser.py

```diff
@@ -28,13 +28,17 @@
 
 def directory_type(string: str) -> Path:
     directory_path = Path(string)
+    # Intentionally passes non-existent directories as valid.
+    # DO NOT CHANGE!
     if directory_path.exists() and not directory_path.is_dir():
         raise ValueError(f"{string} is not a directory.")
     return directory_path
 
 
 def file_type(string: str) -> Path:
     file_path = Path(string)
+    # intentionally passes filepaths that don't exist as valid
+    # DO NOT CHANGE!
```

Comment:
file_type is used for actual files and for paths that should be files at some point, such as log files created when they do not yet exist.

For directory_type, I cannot tell you the reason. I cannot find the same case as above, where a non-existent directory is created on demand.

Not really sure how to test for this. Grepped for all its uses. directory_type is only used to pass a list of dirs to parse and the root directory. For dirs, it is practically ignored if they do not exist but are not created. Passing a non-existent root can cause errors, as sometimes files will be checked if they are relative to it, but again no directory will be created. So maybe this could be changed to not pass non-existent directories.

Reply:
done in 69c69b2e07ce9db14b535d05736360afe80b0ff9